nomad-aws/variables.tf

```diff
@@ -187,3 +187,9 @@ variable "allowed_ips_circleci_server_nomad_access" {
   description = "List of IPv4 ranges that are permitted to access nomad nodes; used for circleci-server-to-nomad communication"
   default     = ["0.0.0.0/0"]
 }
+
+variable "enable_imdsv2" {
+  type        = string
+  description = "Enable or Disable IMDSv2 on Nomad clients. Optional or Required. This is only supported on, or after, CircleCI Server 4.6.0" 
+  default     = "optional"
```

Comment:
I'm thinking we should default to `required` since we version this repo with server releases

Reply:
Would it be smart to do `optional`, cut a new 4.4.x and 4.5.x release, then make it required going forward and do a new 4.6.0 release?

Reply:
IMDSv2 is unsupported in 4.4 and 4.5 so we don't want folks to think they can use it. We already have 4.4 and 4.5 releases that are compatible with those versions of server.

Reply:
Well thats the issue, we explicitly set IMDSv2 to required in our 4.4.1 release of our terraform module 😅 https://github.com/CircleCI-Public/server-terraform/blob/4.4.1/nomad-aws/main.tf#L78-L80, which breaks rerun with SSH on docker for 4.4.x and I presume 4.5.x

Reply:
Oh. Oops. Yeah I guess we need to cherry-pick this. Good call.

Reply:
I think you may want to consider adding a validation block on this, to make sure user set either "optional" or "required"?

https://developer.hashicorp.com/terraform/language/values/variables#custom-validation-rules